app/wizards/placements/add_placement/steps/subject.rb

```diff
@@ -0,0 +1,24 @@
+class Placements::AddPlacement::Steps::Subject
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :school
+  attribute :phase
+  attribute :subject_id
+
+  validates :school, presence: true
+  validates :phase, presence: true
+  validates :subject_id, presence: true
+
+  def subjects_for_selection
+    phase == "Primary" ? Subject.parent_subjects.primary : Subject.parent_subjects.secondary
+  end
```

Comment:
I'm mostly just thinking out loud with this one. I know we've avoided using `case` statements because of the inevitable code coverage issues it causes (i.e. there's always a `nil` branch in the logic).

But instead, what about this?

```suggestion
  def subjects_for_selection
    {
      "Primary" => Subject.parent_subjects.primary,
      "Secondary" => Subject.parent_subjects.secondary,
    }.fetch phase
  end
```

I think I prefer readability of this compared to using a [ternary operator](https://www.rubyguides.com/2019/10/ruby-ternary-operator/), but I'm wondering if it causes the same coverage issues that we get with `case`? 🤔

That said, we do use this approach elsewhere – for example here:

https://github.com/DFE-Digital/itt-mentor-services/blob/4836a5ff5645357884e31f4072207e251f073fc6/app/helpers/routes_helper.rb#L7-L12

So maybe there isn't a code coverage issue. In a way it feels 'safer' because it would crash if an unrecognised phase was passed in.

What do you think? Not a strong opinion, but I'm wondering if it's worth switching to this?